ui/dashboards/src/components/Variables/VariableList.tsx

```diff
@@ -24,51 +24,197 @@ import {
   Table,
   Paper,
   TableHead,
+  Switch,
+  Typography,
+  IconButton,
 } from '@mui/material';
-import { useTemplateVariableDefinitions, useEditMode } from '../../context';
+import { VariableDefinition } from '@perses-dev/core';
+import { useImmer } from 'use-immer';
+import EyeIcon from 'mdi-material-ui/Eye';
+import PencilIcon from 'mdi-material-ui/Pencil';
+import TrashIcon from 'mdi-material-ui/TrashCan';
+
+import { useTemplateVariableDefinitions, useEditMode, useTemplateVariableActions } from '../../context';
 import { TemplateVariable } from './Variable';
+import { VariableEditForm } from './VariableEditorForm';
+
+function VariableEditor(props: {
+  variableDefinitions: VariableDefinition[];
+  onChange: (variableDefinitions: VariableDefinition[]) => void;
+  onCancel: () => void;
+}) {
+  const [variableDefinitions, setVariableDefinitions] = useImmer(props.variableDefinitions);
+  const [variableEditIdx, setVariableEditIdx] = useState<number | null>(null);
+  const isEditingVariable = variableEditIdx !== null;
+
+  const removeVariable = (index: number) => {
+    setVariableDefinitions((draft) => {
+      draft.splice(index, 1);
+    });
+  };
+
+  const addVariable = () => {
+    setVariableDefinitions((draft) => {
+      draft.push({
+        kind: 'TextVariable',
+        spec: {
+          name: 'New Variable',
```

Comment:
The name is supposed to be the thing used to reference the variable in other variable or in the different panels.

So as it's more an ID than just a name, spaces are forbidden.
That's why like for the panel, you have a struct `Display` that contains the name to be displayed.
If `display` is not set, then the name to be displayed is the `ID` of the variable

Reply:
Good call - the follow up PR addresses validation